Environment changes

dopplershift opened the pull request:
- Move to using miniconda
- Use a separate conda environment rather than root
- Put all of the dependencies into a single `environment.yml` with the `conda-forge` channel enabled.

dopplershift committed:
Replace Anaconda with Miniconda.

dopplershift committed:
Change the environment setup.

Use a separate conda environment (rather than root); configure it using
a conda environment file.

dopplershift commented:
from @scollis via email:
>Ok..

>Need this change:


```bash
$conda_bin_path'conda' env create -f /home/ubuntu/AMS_radar_in_the_cloud/ec2_setup/environment.yml
source $conda_bin_path'activate' ams-workshop
```
>Now there is another issue.. for some reason the shell script does not let me source or run files.. 
>for instance the line 
```bash
. .bashrc 
```
>has no impact

>and, more problematic, it seems 

```
source $conda_bin_path'activate' ams-workshop
```

>has not effect..

scollis commented:
Also seeing issue with no numpy found.
-EDIT: This is due to the env not being activated

scollis commented:
Yep.. definite issues with sourcing and changing environment variables in the shell script.. perhaps we need to change it into several scripts..

jjhelmus commented:
Why not use export the change to the PATH inside the script as well as writing it to `.bashrc`

dopplershift commented:
I was working on that as well :grin:

jjhelmus commented on ec2_setup/setup_ec2.sh:
```diff
@@ -10,46 +10,37 @@ echo '--------'
-echo 'PATH="/home/ubuntu/anaconda3/bin:$PATH"' >> .bashrc
+echo 'PATH="/home/ubuntu/miniconda3/bin:$PATH"' >> .bashrc
 
 . .bashrc
```
How about: `export PATH="$HOME/miniconda3/bin:$PATH"`

scollis commented:
This is the lesser of the two issues.. the greater is how to make the source activate take effect?

jjhelmus commented:
> source activate take effect

In the script or in the calling shell?  The former works fine if `activate` is on the path.  The later requires that that `setup_ec2.sh` be sourced not executed.

dopplershift commented:
I think @scollis means in the script. It doesn't seem to be working, since the install of pyart can't find numpy, even here on osx....

scollis commented:
I am trying a few methods.. I have split the script into three coming back into the top script when a source is called.. This works for the path appending (source .bashrc)
but NOT source activate.. Exactly @dopplershift 

Some googling sees others with the same problem but no stackoverflow green ticks..

scollis commented:
I may ping the anaconda list

jjhelmus commented:
Py-ART finds numpy fine in my setup.  Let me clean it up a bit so I can share.

scollis commented:
Great. Thanks @jjhelmus

jjhelmus committed:
Adjust setup_ec2.sh script

export miniconda path
use abtract home directory rather than hard coded
install Py-ART using .tar.gz rather than git checkout for speed

jjhelmus commented:
dopplershift/AMS_radar_in_the_cloud#1

dopplershift committed:
Merge pull request #1 from jjhelmus/conda-forge_jjh

Adjust setup_ec2.sh script

dopplershift commented:
@scollis how about now?

scollis commented:
Just slaying another dragon.. I will test soon.



On 1/17/17 2:47 PM, Ryan May wrote:
>
> @scollis <https://github.com/scollis> how about now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openradar/AMS_radar_in_the_cloud/pull/10#issuecomment-273294699>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAyYB3vc11Z5ULHmA8OO0lFfZgRndEosks5rTSjbgaJpZM4Ll9y5>.
>

scollis commented:
So there is an issue here:

https://github.com/dopplershift/AMS_radar_in_the_cloud/blob/conda-forge/ec2_setup/setup_ec2.sh#L36

on line 26 there is cd ~ so the path is not correct




On 1/17/17 2:47 PM, Ryan May wrote:
>
> @scollis <https://github.com/scollis> how about now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openradar/AMS_radar_in_the_cloud/pull/10#issuecomment-273294699>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAyYB3vc11Z5ULHmA8OO0lFfZgRndEosks5rTSjbgaJpZM4Ll9y5>.
>

dopplershift commented:
FYI, you can go to the "Files Changed" tab above and click on the "+" icon next to the line you want to note and call it out in this PR.

Will look at that.

dopplershift committed:
Fix path to environment file.

Don't change directory to home so early, but stay put so we know where
to find the environment file.

jjhelmus commented:
Do we need to serve this over https?  With can remove all the logic in generating the certs, not have to help users bypass their browsers security, etc if we are alright with password protected http.

scollis commented:
I'm cool with this.. I just boilerplate-ed it from the example


On 1/17/17 3:40 PM, Jonathan J. Helmus wrote:
>
> Do we need to serve this over https? With can remove all the logic in 
> generating the certs, not have to help users bypass their browsers 
> security, etc if we are alright with password protected http.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openradar/AMS_radar_in_the_cloud/pull/10#issuecomment-273308584>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAyYB2AtTBz5hlIf1l9NZ8Xemsd7dq6jks5rTTUxgaJpZM4Ll9y5>.
>

dopplershift commented:
I'm not *super* comfortable with plaintext passwords. But as long as we make it clear that's the case, and that this is ok because we're using a throw-away instance, it might be ok.

Counter-point: We're showing them the way that, in no uncertain terms, they have no business running outside the workshop.

scollis commented:
Perhaps have two .sh files _secure and _unsecure



On 1/17/17 3:43 PM, Ryan May wrote:
>
> I'm not /super/ comfortable with plaintext passwords. But as long as 
> we make it clear that's the case, and that this is ok because we're 
> using a throw-away instance, it might be ok.
>
> Counter-point: We're showing them the way that, in no uncertain terms, 
> they have no business running outside the workshop.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openradar/AMS_radar_in_the_cloud/pull/10#issuecomment-273309346>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAyYB4MEg0En0okvLaNj9lkd6HtcGHI9ks5rTTXkgaJpZM4Ll9y5>.
>

jjhelmus commented:
> not super comfortable with plaintext passwords

Agree, but I think encouraging them to click through browser cert exceptions might be worse.  Plus, I don't want to try figuring out how to do that on IE/Edge which I'm sure someone will have.

dopplershift commented:
As long as we indicate the poor security and that they shouldn't proceed this way as a production use of the notebook, I'm ok.

jjhelmus commented:
Alternatively, we can provide defaults for the cert so that users are not asked those questions and include a `start_notebook.sh` script.  Hosting will be over HTTPS with an unrecognized certificate but would be non-plaintext.

I can have a PR for either option in a few minutes.
